src/voxelalgorithms.h

```diff
@@ -52,7 +52,67 @@ SunlightPropagateResult propagateSunlight(VoxelManipulator &v, VoxelArea a,
 		std::set<v3s16> & light_sources,
 		INodeDefManager *ndef);
 
+/*!
+ * This class iterates trough voxels that intersect with
+ * a line. The collision detection does not see nodeboxes,
+ * every voxel is a cube and is returned.
+ * This iterator steps to all nodes exactly once.
+ */
+struct VoxelLineIterator
+{
+public:
+	//! Starting position of the line in world coordinates.
+	v3f m_start_position;
+	//! Direction and length of the line in world coordinates.
+	v3f m_line_vector;
+	/*!
+	 * Each component stores the next smallest positive number, by
+	 * which multiplying the line's vector gives a vector that ends
+	 * on the intersection of two nodes.
+	 */
+	v3f m_next_intersection_multi;
+	/*!
+	 * Each component stores the smallest positive number, by which
+	 * m_next_intersection_multi's components can be increased.
+	 */
+	v3f m_intersection_multi_inc;
+	/*!
+	 * Direction of the line. Each component can be -1 or 1 (if a
+	 * component of the line's vector is 0, then there will be 1).
+	 */
+	v3s16 m_step_directions;
+	//! Position of the current node.
+	v3s16 m_current_node_pos;
+	//! If true, the next node will intersect the line, too.
+	bool m_has_next;
+
+	/*!
+	 * Creates a voxel line iterator with the given line.
+	 * @param start_position starting point of the line
+	 * in voxel coordinates
+	 * @param line_vector    length and direction of the
+	 * line in voxel coordinates. start_position+line_vector
+	 * is the end of the line
+	 */
+	VoxelLineIterator(const v3f &start_position,const v3f &line_vector);
+
+	/*!
+	 * Steps to the next voxel.
+	 * Updates m_current_node_pos and
+	 * m_previous_node_pos.
+	 * Note that it works even if hasNext() is false,
+	 * continuing the line as a ray.
+	 */
+	void next();
+
+	/*!
+	 * Returns true if the next voxel intersects the given line.
+	 */
+	inline bool hasNext() const{return m_has_next;}
```

Comment:
Sorry, there was no rule for this in the code style guide. I followed the pattern of `map::getSectorsPtr()`. Please include this rule in the guide so other new developers will know how to correctly declare one-line methods.